### PR TITLE
Changed Calendar to add animate, reference, showAdjacentDays

### PR DESCRIPTION
**Pull request by ericsoderberghp**
#### What does this PR do?

Changed Calendar to add animate, reference, onReference, showAdjacentDays, and updateDateRange().
This allows using Calendar in a wider variety of ways.
A new "Dual" Calendar story was added to illustrate linking two Calendars together.
`updateDateRange()` is a new utility function, accessed via `Calendar.updateDateRange()`, that makes it easier to manage the state for a range of dates.

#### Where should the reviewer start?

Calendar/doc.js

#### What testing has been done on this PR?

storybook

#### How should this be manually tested?

storybook

#### Do the grommet docs need to be updated?

yes, to show updateDateRange()

#### Should this PR be mentioned in the release notes?

yes

#### Is this change backwards compatible or is it a breaking change?

backwards compatible

### CODE REVIEW

File: src/js/components/Calendar/doc.js [hunk 1]
```diff
@@ -66,3 +77,18 @@ You can then use that to disable the previous and next buttons.
...
+
+export const docUpdateDateRange = (updateDateRange) => {
+  const DocumentedUpdateDateRange = describe(updateDateRange)
+    .description(`
+      A function to help manage selecting ranges of dates.
+    `).usage(
+      `import { Calendar } from 'grommet';
+const { date, dates, previousSelectedDate } = this.state;
+<Calendar onSelect={d => this.setState(
+  Calendar.updateDateRange(d, { date, dates, previousSelectedDate })
```
[the remark this answers is not in the document]
Comment (ericsoderberghp) [reply src/js/components/Calendar/doc.js]: Agreed. I've changed it to embed it within the Calendar via the `range` property.

File: src/js/components/Calendar/doc.js [hunk 2]
```diff
@@ -53,11 +56,19 @@ You can then use that to disable the previous and next buttons.
 `
     ),
     locale: PropTypes.string.description('The locale to use.'),
+    onReference: PropTypes.func.description(`
```
[the remark this answers is not in the document]
Comment (ericsoderberghp) [reply src/js/components/Calendar/doc.js]: Well, it does align with the `reference` property.

[the remark this answers is not in the document]
Comment (alansouzati) [reply src/js/components/Calendar/doc.js]: I hear you. but does it need to? The reference is not changed yet right? For the fact that we moved months, you may want to reevaluate updating the reference, but you don't need to. This is why I like `onMonthChange` better, or something more descriptive of what is actually happening.

[the remark this answers is not in the document]
Comment (ericsoderberghp) [reply src/js/components/Calendar/doc.js]: Are you thinking we should rename `reference` to `month`?

[the remark this answers is not in the document]
Comment (alansouzati) [reply src/js/components/Calendar/doc.js]: no, I think we can keep reference as is. and `onMonthChange` passes reference just if you are using an uncontrolled Calendar?

```
onMonthChange = (reference) => {} 
```

**Pull request merged**